ballerina/convertor.bal

```diff
@@ -63,13 +63,59 @@ public isolated class Converter {
                 check self.validateTargetRecordName(targetRecordName);
                 JsonToCopybookConverter converter = new (self.schema, targetRecordName);
                 converter.visitSchema(self.schema, readonlyJson);
-                return converter.getValue();
+                return converter.getStringValue();
             }
         } on fail error err {
             return createError(err);
         }
     }
 
+    # Converts the provided record or map<json> value to bytes.
+    # + input - The JSON value that needs to be converted as copybook data
+    # + targetRecordName - The name of the copybook record definition in the copybook. This parameter must be a string
+    # if the provided schema file contains more than one copybook record type definition
+    # + encoding - The encoding of the output bytes array. Default is ASCII
+    # + return - The converted byte array. In case of an error, a `copybook:Error` is is returned
+    public isolated function toBytes(record {} input, string? targetRecordName = (), Encoding encoding = ASCII) returns byte[]|Error {
+        do {
+            readonly & map<json> readonlyJson = check input.cloneWithType();
+            lock {
+                check self.validateTargetRecordName(targetRecordName);
+                JsonToCopybookConverter converter = new (self.schema, targetRecordName);
+                converter.visitSchema(self.schema, readonlyJson);
+                byte[] bytes = check converter.getByteValue();
+                bytes = encoding is EBCDIC ? toEbcdicBytes(bytes) : bytes;
+                return bytes.clone();
```

Comment:
Are we trying to return a readonly byte array here ? If not we could have just do this;
```ballerina
return bytes;
```

Reply:
Returning a value from lock statement.